pkg/kubectl/cmd/update.go

```diff
@@ -42,25 +46,37 @@ Examples:
   $ kubectl update pods my-pod --patch='{ "apiVersion": "v1beta1", "desiredState": { "manifest": [{ "cpu": 100 }]}}'
   <update a pod by downloading it, applying the patch, then updating, requires apiVersion be specified>`,
 		Run: func(cmd *cobra.Command, args []string) {
-			filename := GetFlagString(cmd, "filename")
+			mapper, typer := f.Object(cmd)
+			r := resource.NewBuilder(mapper, typer, ClientMapperForCommand(cmd, f)).
+				ContinueOnError().
```

Comment:
It took me awhile to test this correctly, but yeah, this is a better way, will change that.